### PR TITLE
Rwc2022 challenge restaurant

### PR DESCRIPTION
**Pull request by PetervDooren**

### CODE REVIEW

**Comment by ar13pit** [comment]
I'm picking this one up

**Comment by ar13pit** [comment]
@PetervDooren @MatthijsBurgh Could you guys review the changes in 1be45084ed5987fbe4d995c0ceee939f2191681f and 861500c181a465066ec2328108a3063b5b725ac7 as that is specific to the challenge. Leave the last commit, which is of picovoice as I'll make changes to it.

File: challenge_restaurant/src/challenge_restaurant/restaurant.py [hunk 1]
```diff
@@ -77,6 +73,26 @@ def __init__(self, robot):
...
                                                 'failed': 'WAIT_FOR_CUSTOMER'})
+            #
+            # smach.StateMachine.add('SAY_SPEAK',
+            #                        states.human_interaction.Say(
+            #                            robot,
+            #                            "Please make sure that you speak loudly and directly into my microphone "
+            #                            "AND SPEAK AFTER THE PING. Like is shown "
+            #                            "on my screen", block=True),
+            #                        transitions={'spoken': 'SHOW_IMAGE_SPEAK'})
+            #
```
Comment (PetervDooren) [inline challenge_restaurant/src/challenge_restaurant/restaurant.py:84]: Please remove commented code block

File: challenge_restaurant/src/challenge_restaurant/restaurant.py [hunk 2]
```diff
@@ -102,9 +118,29 @@ def __init__(self, robot):
                                    states.navigation.NavigateToObserve(
                                        robot=robot, entity_designator=customer_designator,
                                        radius=0.8),
-                                   transitions={'arrived': 'TAKE_ORDER',
+                                   transitions={'arrived': 'SAY_SPEAK_2',
                                                 'unreachable': 'SAY_NAVIGATE_TO_CUSTOMER_FALLBACK',
-                                                'goal_not_defined': 'WAIT_FOR_CUSTOMER'})
+                                                'goal_not_defined': 'SAY_WAVING_2'})
+
+            smach.StateMachine.add('SAY_SPEAK_2',
```
Comment (PetervDooren) [inline challenge_restaurant/src/challenge_restaurant/restaurant.py:125]: suggestion:
SAY_INSTRUCT_CUSTOMER
may be a more informative state name.

File: challenge_restaurant/src/challenge_restaurant/take_orders.py [hunk 3]
```diff
@@ -66,20 +69,18 @@ def execute(self, userdata=None):
...
                     speech_result = self._robot.hmi.query(description="Can I please take your order",
-                                                          grammar=knowledge.order_grammar, target="O")
+                                                        grammar=knowledge.order_grammar, target="O")
                     break
                 except TimeoutException:
-                    if count < 5:
-                        self._robot.speech.speak(random.choice(["I'm sorry, can you repeat",
-                                                                "Please repeat your order, I didn't hear you",
-                                                                "I didn't get your order, can you repeat it",
-                                                                "Please speak up, as I didn't hear your order"]))
```
Comment (PetervDooren) [inline challenge_restaurant/src/challenge_restaurant/take_orders.py:82]: I would prefer this option as it makes the robot not repeat itself. I get the choice for the single command in a robocup scenario but I prefer this one for a general waiter-robot

File: challenge_restaurant/src/challenge_restaurant/take_orders.py [hunk 4]
```diff
@@ -135,6 +136,10 @@ def execute(self, userdata=None):
 
         self._robot.speech.speak(sentence)
 
+        if "water" in self._orders:
+            water_sentence = "I am too weak to carry the water, please bring that yourself. And please make me stronger."
+            self._robot.speech.speak(water_sentence)
+
```
Comment (PetervDooren) [inline challenge_restaurant/src/challenge_restaurant/take_orders.py:172]: I think we would consider this a Robocup hack. Not sure if it is smart to add this to the code. 
On the one hand Hero does not get stronger. But it is dependent on the exact bottle of water being used. Furthermore there may be other items we need to exclude based on their weight and this implementation is not configurable.

**Comment by PetervDooren** [comment]
I looked at the changes in https://github.com/tue-robotics/tue_robocup/commit/1be45084ed5987fbe4d995c0ceee939f2191681f and https://github.com/tue-robotics/tue_robocup/commit/861500c181a465066ec2328108a3063b5b725ac7 only as specified by @ar13pit

**Pull request closed without merging**